libs/evcs-service/src/main/java/uk/gov/di/ipv/core/library/service/EvcsService.java

```diff
@@ -109,10 +109,12 @@ private void persistEvcsUserVCs(
                                         existingEvcsUserVCs.stream()
                                                 .noneMatch(
                                                         evcsVC ->
-                                                                evcsVC.vc()
-                                                                        .equals(
-                                                                                credential
-                                                                                        .getVcString())))
+                                                                (evcsVC.vc()
+                                                                                .equals(
+                                                                                        credential
+                                                                                                .getVcString())
+                                                                        && evcsVC.state()
+                                                                                .equals(CURRENT))))
```

Comment:
I think this is not _quite_ right - this will recreate the VC, but instead you want to update it to `CURRENT`.

Instead, there might be a new category in the updates:
|  | CURRENT | PENDING |
| ------------- | ------------- | ------------- |
| In session  | No change  | Update to CURRENT |
| Not in session  | Update to HISTORIC  | Update to ABANDONED |

At the moment I think we're doing the bottom row, but not the top-right.

Reply:
yes we are doing the bottom one for update.

Reply:
I think what you've implemented is slightly different: you're updating the existing record to ABANDONED, and then creating a new CURRENT record. Instead I think we should just update it directly to CURRENT?

To expand the table out in all cases:

When storing a current identity:
| EVCS | CURRENT | PENDING | not present |
| ------------- | ------------- | ------------- | ------------- |
| In session  | No change  | Update to CURRENT | Create new CURRENT |
| Not in session  | Update to HISTORIC  | Update to ABANDONED | - |

When storing a pending identity:
| EVCS | CURRENT | PENDING | not present |
| ------------- | ------------- | ------------- | ------------- |
| In session  | -  | - | Create new PENDING |
| Not in session  | -  | Update to ABANDONED | - |